### PR TITLE
bases: pre-installation of common packages

### PR DESCRIPTION
**Pull request by syu-w**
All images will be pre-installed with `build-essential` (or equivalent), `git`, and `python3` (with `devel`, `pip`, `venv`, `setuptools`, `wheel`)

### CODE REVIEW

**Comment by lengau** [comment]
Thinking purely in a `craft-providers` context, are these packages actually necessary?  I could imagine that the Python ones might be (if, for example, craft-providers runs Python code inside the provider), but these seem more application-specific otherwise.

**Comment by syu-w** [comment]
Those packages (gcc, etc.) are actually pre-installed in the `buildd` image. So if only use that image, you don't need to add them. However, since we start support more general images, I think they should contain the same base components as buildd.

Also add them into base image will reduce the time spent on repeated downloads and installations those base packages.

File: craft_providers/bases/ubuntu.py [hunk 1]
```diff
@@ -97,7 +97,7 @@ def __init__(
 
         self._set_hostname(hostname)
 
-        self._packages = ["apt-utils", "curl", "fuse", "udev"]
+        self._packages = ["apt-utils", "build-essential", "curl", "fuse", "udev"]
```
Comment (mr-cal) [inline craft_providers/bases/ubuntu.py:100]: This is already installed on `buildd` images but installs extra packages on the `daily` interim images we are currently using?

Comment (syu-w) [reply craft_providers/bases/ubuntu.py:100]: Yes, and apt will not try to install it again if it already installed.

Comment (mr-cal) [reply craft_providers/bases/ubuntu.py:100]: Great!

**Review by mr-cal — APPROVED** [review]
I'm fine with all images matching `buildd`'s by having `build-essentials` and `python` pre-installed.  

Since Alex has concerns, will you wait to merge until you come to an agreement with him?

**Comment by lengau** [comment]
Ideally I'd prefer for this to be the application's responsibility, since that also allows the app to do other preconfiguration that applies to all projects for that application (or even better — multiple snapshots for the same application depending on context). However, the perfect being the enemy of the good and all, now that I understand the underlying goal I'm going to withdraw my concerns since this still makes a whole lotta sense. Thanks for explaining! :-)

**Review by lengau — APPROVED** [review]

**Comment by syu-w** [comment]
@mr-cal @lengau I added `use_default_packages` in case others want to override the default packages.

**Review by lengau — APPROVED** [review]

**Review by mr-cal — APPROVED** [review]

**Pull request merged**